activerecord/lib/active_record/connection_adapters/mysql/quoting.rb

```diff
@@ -18,9 +18,9 @@ def cast_bound_value(value)
           when BigDecimal
             value.to_s("F")
           when true
-            "1"
+            "TRUE"
```

Comment:
Are you sure this does what you think? If you write a test I think you'll see it will be serialized as a String and not a Boolean.